pkg/volume/gce_pd/gce_util.go

```diff
@@ -122,6 +112,135 @@ func (util *GCEDiskUtil) DetachDisk(pd *gcePersistentDisk) error {
 	if err := gce.(*gce_cloud.GCECloud).DetachDisk(pd.pdName); err != nil {
 		return err
 	}
+
+	return waitForDiskToDetach(pd)
+}
+
+// Wait for the disk device to be created
+func waitForDiskToAttach(pd *gcePersistentDisk, sdBeforeSet util.StringSet) (string, error) {
+	devicePaths := getDiskByIdPaths(pd)
+	for numTries := 0; numTries < 10; numTries++ {
+		if err := udevadmChangeToNewDrives(sdBeforeSet); err != nil {
+			// udevadm errors should not block disk attachment, log and continue
+			glog.Errorf("%v", err)
+		}
+
+		for _, path := range devicePaths {
```

Comment:
More robust to watch all of them, and grab on to the first one that shows up. And then as long as one shows up we know that the device is successfully attached and can use that as the global mount point.